### PR TITLE
Update editor.tsx

### PR DESCRIPTION
**Pull request by Zhark10**
I created custom hooks! It needs to be transferred to other files, but I didn't have time.

By the way, this garbage can also be put in hooks if you want:))
const updatedPolygonsState = [...state.polygons];

### CODE REVIEW

File: src/components/containers/editor/editor.tsx [hunk 1]
```diff
@@ -12,25 +12,33 @@ import PolygonsLayer from './components/polygons-layer';
 
 import EditorContext from './state/context';
 
-const EditorContainer: React.FC = () => {
-  const { state, setMousePosition, setPlanImage, toggleTool, updatePolygons } = React.useContext(EditorContext);
-  const [isMouseOverStartPoint, toggleMouseOverStartPoint] = React.useState(false);
+// use hooks
```
Comment (AlexMubarakshin) [inline src/components/containers/editor/editor.tsx]: Remove comments, pls

File: src/components/containers/editor/editor.tsx [hunk 2]
```diff
@@ -12,25 +12,33 @@ import PolygonsLayer from './components/polygons-layer';
...
 
-  const onChangeImageClick = async () => {
-    const image = await loadImageFromFS();
-    setPlanImage(image);
-  };
+const useDragHook = (updatedPolygonsState: any) => {
```
Comment (AlexMubarakshin) [inline src/components/containers/editor/editor.tsx]: No needs to pass polygonState to `useDragHook` and `useStageHook` hooks.

Comment (AlexMubarakshin) [reply src/components/containers/editor/editor.tsx]: And rename like to `usePolygonHandlers`, `usePolygon` or something like this :)

File: src/components/containers/editor/editor.tsx [hunk 3]
```diff
@@ -12,25 +12,33 @@ import PolygonsLayer from './components/polygons-layer';
...
+      ...updatedPolygonsState[polygonIndex].points.slice(index + 1)
+    ];
+
+    updatePolygons(updatedPolygonsState);
+  };  
+
+  return { onDragPoint };
+}
 
+const useStageHook = (updatedPolygonsState: any) => {
```
Comment (AlexMubarakshin) [inline src/components/containers/editor/editor.tsx]: Rename to `useSceneHandlers`

File: src/components/containers/editor/editor.tsx [hunk 4]
```diff
@@ -61,39 +69,46 @@ const EditorContainer: React.FC = () => {
     toggleMouseOverStartPoint(overPoint);
   }
 
-  const onDragPoint = (event: Konva.KonvaEventObject<DragEvent>, polygonIndex: number, pointIndex: number) => {
-    const updatedPolygonsState = [...state.polygons];
-
-    const index = event.target.index - 1;
+  return { onStageClick, onMouseOverStartPointOfPolygon };
```
Comment (AlexMubarakshin) [inline src/components/containers/editor/editor.tsx]: `onMouseOverStartPointOfPolygon` must be the part of `usePolygonHandlers` hook, on 17 line

File: src/components/containers/editor/editor.tsx [hunk 5]
```diff
@@ -61,39 +69,46 @@ const EditorContainer: React.FC = () => {
...
+// component
 
-    updatedPolygonsState[polygonIndex].points = [
-      ...updatedPolygonsState[polygonIndex].points.slice(0, index),
-      pos,
-      ...updatedPolygonsState[polygonIndex].points.slice(index + 1)
-    ];
+const EditorContainer: React.FC = () => {
+  const { state, setMousePosition, setPlanImage, toggleTool } = React.useContext(EditorContext);
+  const updatedPolygonsState = [...state.polygons];
```
Comment (AlexMubarakshin) [inline src/components/containers/editor/editor.tsx]: There is no need to create `updatedPolygonsState` variable, hooks on below lines have access to state.

File: src/components/containers/editor/editor.tsx (hunk 1 above)
Comment (Zhark10) [reply src/components/containers/editor/editor.tsx]: yup

File: src/components/containers/editor/editor.tsx (hunk 2 above)
Comment (Zhark10) [reply src/components/containers/editor/editor.tsx]: First of all, I wrote this in a comment right away. I didn`t have time to change))

File: src/components/containers/editor/editor.tsx (hunk 5 above)
Comment (Zhark10) [reply src/components/containers/editor/editor.tsx]: Yeah, i know about it! I wrote above.. :)

File: src/components/containers/editor/editor.tsx (hunk 4 above)
Comment (Zhark10) [reply src/components/containers/editor/editor.tsx]: I don't agree. `isMouseOverStartPoint` is used as a flag for `isPolygonFinished` inside `useSceneHandlers`

File: src/components/containers/editor/editor.tsx (hunk 3 above)
Comment (Zhark10) [reply src/components/containers/editor/editor.tsx]: I agree

**Pull request merged**